src/sentry/static/sentry/app/views/eventsV2/saveQueryButton.tsx

```diff
@@ -66,6 +69,21 @@ class EventsSaveQueryButton extends React.Component<Props, State> {
         pathname: location.pathname,
         query: view.generateQueryStringObject(),
       });
+
+      const eventKeyName = editingExistingQuery
+        ? {
+            eventKey: 'discover_v2.save_existing_query',
+            eventName: 'Discoverv2: Save a saved query as a new query',
+          }
+        : {
+            eventKey: 'discover_v2.save_new_query',
+            eventName: 'Discoverv2: Save a new query',
+          };
+      trackAnalyticsEvent({
+        ...eventKeyName,
+        organization_id: organization.id,
+        ...extractAnalyticsQueryFields(payload),
+      });
```

Comment:
We should track the event asap (which is before the method calls an API endpoint), even if it comes at a perf cost. I think that's an acceptable tradeoff at this point.

 For this implementation, we will lose the event if the API call fails and that will cause inaccurate data.

IMO, we should split this into 3 events: `discover_v2.save_existing_query_request`, `discover_v2.save_existing_query_success`, `discover_v2.save_existing_query_fail` so we can have some insights even when something went wrong for weird reason (e.g. failed requests because they did not provide a query.name may indicate that our UI was unclear).

Reply:
mmm gotcha that makes sense. Should they all have the same information or should only the success have the full info?

Reply:
They can have the same info but with different keys.

Reply:
actually is it necessary to do 3 events? wouldn't 2 provide the same information because a request will always lead to success/fail (unless the frontend crashes midway through)

Reply:
We want to know for sure that `success + fail == requests`. 

If there's some discrepancies, it might indicate that an issue exists (e.g. requests might be canceled because they take forever to load, which is a useful sign of UX issues).

Reply:
Can we have this as a  `success` prop within the `discover_v2.save_existing_query_request` event? 

So that we don't have 3 distinct events.

Reply:
you got it